LONDON CLASS_8-MARGARITA_GLUSHKOVA-JS2-WEEK_3

RitaGlushkova opened the pull request:
<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

alinosratipour commented on mandatory/3-slideshow/slideshow.js:
```diff
@@ -1 +1,76 @@
+  "https://images.unsplash.com/photo-1576201836106-db1758fd1c97?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGRvZ3xlbnwwfHwwfA%3D%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
+  "https://images.unsplash.com/photo-1534361960057-19889db9621e?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MjV8fGRvZ3xlbnwwfHwwfA%3D%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60",
+  "https://images.unsplash.com/photo-1502673530728-f79b4cab31b1?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxzZWFyY2h8NDV8fGRvZ3xlbnwwfHwwfA%3D%3D&auto=format&fit=crop&w=500&q=60",
+  "https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxzZWFyY2h8NjV8fGRvZ3xlbnwwfHwwfA%3D%3D&auto=format&fit=crop&w=500&q=60",
```
Hello Daniel your code is very clean and easy to read and your solution is very smart looking.
only suggestion i have to save some image into your project to avoid long URL inside imgList = [] array.that wonderful job well done.

DanGhanbari replied:
Thanks, Ali for your feedback.

muratdemirtas2001 commented on mandatory/1-alarmclock/alarmclock.js:
```diff
@@ -1,4 +1,43 @@
+    let color1 = Math.floor(Math.random() * 256);
+    let color2 = Math.floor(Math.random() * 256);
+    let color3 = Math.floor(Math.random() * 256);
+    let RGB = `rgb(${color1}, ${color2}, ${color3})`;
```
Hi Daniel, it is a great way to  produce random color with RGB

Jpsi commented on mandatory/1-alarmclock/alarmclock.js:
```diff
@@ -1,4 +1,43 @@
-function setAlarm() {}
+function setAlarm() {
```
Well done for making the alarm clock work, and especially for getting the background to flash in different colours! 🙂

If I were to recommend one thing, it would be  to break your code down into functions - this will make it more structured, more readable, and cleaner.

For example, instead of using an anonymous function for the first argument for `setInterval`, you can define it outside of `setAlarm`, give it a name like `decrementAndRenderTimer`, and pass it to `setInterval`.

I would do that repeatedly until functions are no longer than about 4-5 lines long, and - more importantly - each function accomplishes one specific thing that can be described with the function name.

Other than that, notice that when the "Set Alarm" button is pressed multiple times while the timer is running, the timer starts showing the wrong time. This happens because `setInterval` gets called multiple times without calling `clearInterval`.

DanGhanbari replied:
Thank you Philip for your feedback. It was a very helpful and good point.

RitaGlushkova committed:
A - exercise (complete)

RitaGlushkova committed:
B-C exercises complete

RitaGlushkova committed:
2-project only main part complete.

Still need to do advanced section and extra advanced

RitaGlushkova committed:
2-projects deleted btn added

I need to re-write this exercise using array. At the moment I am directly adding element to HTM and then requesting values back. It is not working properly

RitaGlushkova commented:
@DiegoJunior0 Hi Henry, I created the pull request, but my 2-project exercise is not using provided array. I need to remake it. 
Maybe we could look at it together sometime next week?

DiegoJunior0 commented:
Good work, yes we can discuss in our call this week on Thursday
